InteractiveHtmlBom/ecad/schema/genericjsonpcbdata_v1.schema

```diff
@@ -0,0 +1,574 @@
+                        "additionalProperties": false,
+                        "properties": {
+                            "layer": { "$ref": "#/definitions/Layer" },
+                            "drawing": { "$ref": "#/definitions/Drawing" }
+                        },
+                        "required": [ "layer", "drawing" ]
+                    }
+                },
+                "layer": {
+                    "$ref": "#/definitions/Layer"
+                }
+            },
+            "minProperties": 6,
+            "title": "Footprint"
+        },
+        "Bbox": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "pos": {
+                    "$ref": "#/definitions/XY"
+                },
+                "relpos": {
+                    "$ref": "#/definitions/XY"
+                },
+                "size": {
+                    "$ref": "#/definitions/XY"
+                },
+                "angle": {
+                    "type": "number"
+                }
+            },
+            "minProperties": 4,
+            "title": "Bbox"
+        },
+        "Pad": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "layers": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/Layer"
+                    },
+                    "minItems": 1,
+                    "maxItems": 2
+                },
+                "pos": {
+                    "$ref": "#/definitions/XY"
+                },
+                "size": {
+                    "$ref": "#/definitions/XY"
+                },
+                "angle": {
+                    "type": "number"
+                },
+                "shape": {
+                    "$ref": "#/definitions/Shape"
+                },
+                "svgpath": { "type": "string" },
+                "polygons": { "$ref": "#/definitions/Polygons" },
+                "radius": { "type": "number" },
+                "chamfpos": { "type": "integer" },
+                "chamfratio": { "type": "number" },
+                "type": {
+                    "$ref": "#/definitions/PadType"
+                },
+                "pin1": {
+                    "type": "integer", "minimum": 1, "maximum": 1
+                },
+                "drillshape": {
+                    "$ref": "#/definitions/Drillshape"
+                },
+                "drillsize": {
+                    "$ref": "#/definitions/XY"
+                },
+                "offset": {
+                    "$ref": "#/definitions/XY"
+                },
+                "net": { "type": "string" }
+            },
+            "required": [
+                "layers",
+                "pos",
+                "size",
+                "angle",
+                "shape",
+                "type"
+            ],
```

Comment:
This is a bit complicated. If svgpath is present then pos, size, angle are not required. Please use if/then clause.
Also if type is custom then polygons is required.